### PR TITLE
(rebase) Namespace writing - resolves #52

### PR DESCRIPTION
**Pull request by lostapathy**
This rebases #52 to add support for writing namespaces.

Like the original requester I'm open to feedback on this, I just wanted to get that PR updated and ready to either merge, ready for feedback on what else to do, or else reject the whole idea.

### CODE REVIEW

File: spec/xml/namespace_spec.rb [hunk 1]
```diff
@@ -50,9 +50,9 @@ class VApp
     describe "#to_xml" do
       it "should reproduce the input xml" do
         output = ROXML::XML::Document.new
-        output.root = VApp.from_xml(@xml).to_xml
-        pending "Full namespace write support"
-        output.should == ROXML::XML.parse_string(@xml)
+        namespaces = {'vmw' => "http://foo.example.com", 'gronk' => "http://gronk.example.com"}
```
Comment (Empact) [inline spec/xml/namespace_spec.rb]: This variable isn't referenced

File: lib/roxml.rb [hunk 2]
```diff
@@ -42,6 +42,7 @@ def to_xml(params = {})
             ref.update_xml(root, value)
           end
         end
+        params[:namespaces].each { |prefix, url| root.add_namespace_definition(prefix, url) } if params[:namespaces]
```
Comment (Empact) [inline lib/roxml.rb]: Could you shorten this line? It looks on the long side...

**Comment by lostapathy** [comment]
Good catches - thanks for the review

**Comment by lostapathy** [comment]
It turns out there was a logical issue in #32 - which caused empty elements without a default to get skipped in the output unless they had a default.  I added a failing test of this, confirm that test passed prior to #32, then updated the code added on #32 to make it pass again.

**Comment by lostapathy** [comment]
Sorry for the noise - I reset my branch, rebased and reapplied the commits and I think this is ready to look at now.

**Pull request merged**